staging/src/kubevirt.io/api/core/v1/types.go

```diff
@@ -100,7 +100,15 @@ type VirtualMachineInstanceSpec struct {
 	SchedulerName string `json:"schedulerName,omitempty"`
 	// If toleration is specified, obey all the toleration rules.
 	Tolerations []k8sv1.Toleration `json:"tolerations,omitempty"`
-
+	// TopologySpreadConstraints describes how a group of VMIs will be spread across a given topology
+	// domains. K8s scheduler will schedule VMI pods in a way which abides by the constraints.
+	// +optional
+	// +patchMergeKey=topologyKey
+	// +patchStrategy=merge
+	// +listType=map
+	// +listMapKey=topologyKey
+	// +listMapKey=whenUnsatisfiable
```

Comment:
Out of curiosity: why is there only one `patchMergeKey` (i.e. topologyKey) versus two `listMapKey` (i.e. topologyKey and whenUnsatisfiable)

Based on my understanding from here https://ymmt2005.hatenablog.com/entry/2020/04/14/An_example_of_using_dynamic_client_of_k8s.io/client-go  having multiple listMapKey means if the topologyKey is same, but whenUnsatisfiable value is different, server side apply will be able to handle that vs it will not work in patch merge on client side.

But having two different `whenUnsatisfiable` with same `topologyKey` makes scheduling hard especially when they are ANDed.

Can you please help me understand what am I missing?

Reply:
> But having two different whenUnsatisfiable with same topologyKey makes scheduling hard especially when they are ANDed.

This should purely have an effect on patch operations. It has no impact on the scheduler. It is a 1:1 copy from k8s therefor it is probably fine. I can't say I understand it though :D

Reply:
if this is a 1:1 copy than it should be fine, thanks for checking @rmohr